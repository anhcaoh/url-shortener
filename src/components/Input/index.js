import React, { useState } from "react";
import "./input.scss";
const Input = (props) => {
    const { type, placeholder, style, className, defaultValue, handleOnChange, autoFocus } = props;
    const [value, setValue ] = useState( defaultValue || "" );
    const onChangeHanlder = (e) => {
        const _value = e.target.value;
        setValue( _value );
        handleOnChange && handleOnChange( _value );
    };
    return (<input
    type={ type || "text"}
    autoComplete="new-password"
    autoFocus={ autoFocus }
    placeholder={placeholder}
    style={style}
    className={className}
    value={ value } 
    onChange={ onChangeHanlder }
    />);
};
export default Input;