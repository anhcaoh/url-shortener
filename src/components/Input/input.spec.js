import React from "react";
import renderer from "react-test-renderer";
import Input from "./index";
import { mount } from "enzyme";

describe("Input", () => {
    it("should render snapshot", () => {
        const component = renderer.create(<Input />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot(); 
    });
    it("should update state when onChange input", () => {
        let wrapper = mount(<Input />);
        console.log(wrapper.find("input").props().value);
        const _newValue = "Input value has changed.";
        const eventObj = { target: {value: _newValue }};
        wrapper.find("input").simulate("change",  eventObj );
        wrapper.update();
        expect( wrapper.find("input").props().value ).toEqual( _newValue );
    });
});