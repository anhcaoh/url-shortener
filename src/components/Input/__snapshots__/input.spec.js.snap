// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render snapshot 1`] = `
<input
  autoComplete="new-password"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
